frontend/src/metabase/entities/containers/EntityForm.jsx

```diff
@@ -24,10 +24,11 @@ export default class EntityForm extends React.Component {
         onSubmit={values =>
           values.id != null ? update(values) : create(values)
         }
-        onSubmitSuccess={action =>
+        onSubmitSuccess={action => {
+          console.log("submit success", action);
           // maybe eventually pass the whole object instead of just the id?
-          onSaved && onSaved({ id: action.payload.result })
-        }
+          return onSaved && onSaved(action.payload);
```

Comment:
I did end up doing the `Todo` here so that some of the user creation flow modals would have access to more than just the ID after creation.